src/core_plugins/metrics/public/components/aggs/math.js

```diff
@@ -0,0 +1,93 @@
+import React, { Component, PropTypes } from 'react';
+import _ from 'lodash';
+import uuid from 'uuid';
+import AggRow from './agg_row';
+import AggSelect from './agg_select';
+
+import createChangeHandler from '../lib/create_change_handler';
+import createSelectHandler from '../lib/create_select_handler';
+import createTextHandler from '../lib/create_text_handler';
+import Vars from './vars';
+
+class MathAgg extends Component {
+
+  componentWillMount() {
+    if (!this.props.model.variables) {
+      this.props.onChange(_.assign({}, this.props.model, {
+        variables: [{ id: uuid.v1() }]
+      }));
+    }
+  }
+
+  render() {
+    const { siblings } = this.props;
+
+    const defaults = { script: '' };
+    const model = { ...defaults, ...this.props.model };
+
+    const handleChange = createChangeHandler(this.props.onChange, model);
+    const handleSelectChange = createSelectHandler(handleChange);
+    const handleTextChange = createTextHandler(handleChange);
+
+    return (
+      <AggRow
+        disableDelete={this.props.disableDelete}
+        model={this.props.model}
+        onAdd={this.props.onAdd}
+        onDelete={this.props.onDelete}
+        siblings={this.props.siblings}
+      >
+        <div className="vis_editor__row_item">
+          <div>
+            <div className="vis_editor__label">Aggregation</div>
+            <AggSelect
+              siblings={this.props.siblings}
+              value={model.type}
+              onChange={handleSelectChange('type')}
+            />
+            <div className="vis_editor__variables">
+              <div className="vis_editor__label">Variables</div>
+              <Vars
+                metrics={siblings}
+                onChange={handleChange}
+                name="variables"
+                model={model}
+                includeSiblings={true}
+              />
+            </div>
+            <div className="vis_editor__row_item">
+              <div className="vis_editor__label">
+                Basic math (see <a href="http://mathjs.org/docs/expressions/syntax.html" target="_blank">MathJS</a>) - Variables
+                are keys on the <code>params</code> object, i.e. <code>params.&lt;name&gt;</code> To access all the data use
+                <code>params._all.&lt;name&gt;.values</code> for an array of the values and <code>all.&lt;name&gt;.timestamps</code>
+                for an array of the timestamps.  <code>params._timestamp</code> is available for the current bucket&apos;s timestamp
+                and <code>params._index</code> is available for the current bucket&apos;s index.
+              </div>
+              <input
```

Comment:
This input is missing a [correct label](https://github.com/elastic/kibana/blob/master/style_guides/accessibility_guide.md#label-every-form-element).